relayer/chains/substrate/event_parser.go

```diff
@@ -0,0 +1,236 @@
+package substrate
+
+import (
+	"encoding/json"
+	"strconv"
+
+	rpcclienttypes "github.com/ComposableFi/go-substrate-rpc-client/v4/types"
+	clienttypes "github.com/cosmos/ibc-go/v5/modules/core/02-client/types"
+	"github.com/cosmos/relayer/v2/relayer/processor"
+	"github.com/cosmos/relayer/v2/relayer/provider"
+	"go.uber.org/zap/zapcore"
+)
+
+// ibcMessage is the type used for parsing all possible properties of IBC messages
+type ibcMessage struct {
+	eventType string
+	info      ibcMessageInfo
+}
+
+type ibcMessageInfo interface {
+	MarshalLogObject(enc zapcore.ObjectEncoder) error
+}
+
+// ibcMessagesFromTransaction parses all events within a transaction to find IBC messages
+func (scp *SubstrateChainProcessor) handleIBCMessagesFromEvents(ibcEvents rpcclienttypes.IBCEventsQueryResult, height uint64, c processor.IBCMessagesCache) error {
+	for i := 0; i < len(ibcEvents); i++ {
+		for eType, data := range ibcEvents[i] {
+			var info ibcMessageInfo
+			var eventType string
+
+			marshalled, _ := json.Marshal(data)
+
+			switch eType {
+
+			case "CreateClient":
```

Comment:
It looks like a `fallthrough` was intended here. In go, cases will not fall through by default.

These can be simplified to, for example:
`case "CreateClient", "UpgradeClient", "ClientMisbehaviour":`

Reply:
Done